child_compassion/wizard/project_description_de.py

```diff
@@ -0,0 +1,175 @@
+                      project.name, project.community_name,
+                      project.distance_from_closest_city,
+                      project.country_common_name,
+                      project.community_name,
+                      "" if not terrain_desc[0] else
+                      " einer " + terrain_desc[0],
+                      project.community_population))
+
+        return string
+
+    @classmethod
+    def _gen_build_mat_de(cls, cr, uid, project, context=None):
+        """ Generate house build materials, there are no specificities
+            in this part
+        """
+        floor_mat = [mat.value_de if mat.value_de else mat.value_en
+                     for mat in project.floor_material_ids]
+
+        wall_mat = [mat.value_de if mat.value_de else mat.value_en
+                    for mat in project.wall_material_ids]
+
+        roof_mat = [mat.value_de if mat.value_de else mat.value_en
+                    for mat in project.roof_material_ids]
+
+        string = u"Die Häuser sind typischerweise mit "
+
+        if wall_mat[0] and floor_mat[0] and roof_mat[0]:
+            string += (u"%s gebaut und haben %s, sowie %s. " % (
+                       wall_mat[0], floor_mat[0], roof_mat[0]))
+
+        elif ((wall_mat[0] and floor_mat[0]) or (wall_mat[0] and
+              roof_mat[0]) or (floor_mat[0] and roof_mat[0])):
+            if not wall_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           floor_mat[0], roof_mat[0]))
+            elif not floor_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           wall_mat[0], roof_mat[0]))
+            elif not roof_mat[0]:
+                string += (u"%s gebaut und haben %s. " % (
+                           wall_mat[0], floor_mat[0]))
+
+        elif (wall_mat[0] or floor_mat[0] or roof_mat[0]):
+            if wall_mat[0]:
+                string += u"%s gebaut. " % wall_mat[0]
+            elif floor_mat[0]:
+                string += u"%s gebaut. " % floor_mat[0]
+            elif roof_mat[0]:
+                string += u"%s gebaut. " % roof_mat[0]
+        else:
+            string = ""
+
+        return string
+
+    @classmethod
+    def _gen_primary_diet_de(cls, cr, uid, project, context=None):
+        """ Generate spoken languages(s) and primary diet, there are
+            no specificities in this part
+        """
+        primary_diet = [diet.value_de if diet.value_de else diet.value_en
+                        for diet in project.primary_diet_ids]
+
+        spoken_languages = [lang.value_de if lang.value_de else lang.value_en
+                            for lang in project.spoken_languages_ids]
+
+        string = (u"Die ethnischen Merheiten %s %s" % (
+                  u"sind" if len(spoken_languages) > 1 else u"ist",
+                  cls._gen_list_string(spoken_languages, ', ', ' und ')))
+
+        if spoken_languages[0]:
```

Comment:
Replace with len(spoken_languages). Otherwise, if spoken_languages is empty, spoken_languages[0] will raise an exception.